src/sscg_generator.erl

```diff
@@ -1,68 +1,64 @@
-% @doc Main application and CLI entry point.
 -module(sscg_generator).
+-moduledoc "Main application and CLI entry point".
 
 -behaviour(application).
 -behaviour(cli).
 
 -define(REQUIRED_OTP_VERSION, 26).
 
 % Callbacks
--export([main/1]).
--export([cli/0]).
-
--export([start/2]).
--export([stop/1]).
+-export([main/1, cli/0]).
+-export([start/2, stop/1]).
 
 %--- Callbacks -----------------------------------------------------------------
 
-% @doc Main CLI entry point.
+%--- Callbacks: Application
+-doc "Main CLI entry point".
+-spec main(Args :: [string()]) -> term() | no_return().
 main(Args) ->
-    check_otp_version(?REQUIRED_OTP_VERSION),
-    {ok, _} = application:ensure_all_started(sscg_generator),
-    cli:run(Args, #{
-        progname => ?MODULE,
-        modules => [?MODULE,
-                    sscg_generator_generate,
-                    sscg_generator_publish],
-        warn => false
-    }).
+    case ensure_minimum_otp_version(?REQUIRED_OTP_VERSION) of
+        ok ->
+            {ok, _} = application:ensure_all_started(sscg_generator),
+            Options = #{progname => ?MODULE,
+                        modules  => [?MODULE,
+                                     sscg_generator_generate,
+                                     sscg_generator_publish],
+                        warn     => suppress},
+            cli:run(Args, Options);
```

Comment:
It looks like the `cli` module is undocumented -- you shouldn't use it. You probably want `argparse:run/2` instead, as shown [here](https://www.erlang.org/doc/apps/stdlib/argparse.html#module-quick-start).

Reply:
Mm, I didn't find `:run/2`.

I actually liked how cli:run/2 had the modules set up (though it’s odd that the main functionality info was in the run_options() map), with one file/module per command and all the logic written there. 

```erlang 
-type run_options() :: #{
    modules => all_loaded | module() | [module()],
    warn => suppress | warn,
    help => boolean(),
    error => ok | error | halt | {halt, non_neg_integer()},
    prefixes => [integer()],%% prefixes passed to argparse
    %% default value for all missing not required arguments
    default => term(),
    progname => string() | atom()   %% specifies executable name instead of 'erl'
}.
```

https://github.com/max-au/argparse/blob/master/src/cli.erl 

> Warning This project is no longer maintained, because argparse is now a part of Erlang/OTP: https://www.erlang.org/doc/man/argparse.html (starting with OTP 26).

Reply:
Sorry, I meant `argparse:run/3`. https://www.erlang.org/doc/apps/stdlib/argparse.html#run/3

Reply:
It's unfortunate that there is also an external library called argparse. Can you work with the OTP one instead?

Reply:
And I'm just realizing that you are using the third-party library, sorry... 😅
In this case feel free to continue using it, as well as `cli:run/2`.